pkg/finality-grandpa/lib.go

```diff
@@ -0,0 +1,398 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"github.com/tidwall/btree"
+	"golang.org/x/exp/constraints"
```

Comment:
Do we have plans to update our go version?

Reply:
Yes, we should discuss it with the team. #3438 is trying to do this, but @kishansagathiya had a comment that some dependencies may not work